ssr/src/page/token/info.rs

```diff
@@ -144,20 +152,21 @@ fn TokenInfoInner(
                 </div>
                     <Show when= move || is_user_principal>
                         <a
-                            href=format!("/token/transfer/{root}")
+                            href=format!("/token/transfer/{}", root.map(|r| r.to_text()).unwrap_or(meta_c1.name.to_lowercase()))
                             class="fixed bottom-20 left-4 right-4 p-3 bg-primary-600 text-white text-center md:text-lg rounded-full z-50"
                         >
                             Send
                         </a>
                     </Show>
                 {if key_principal.is_some() {
-                    view! { <Transactions source=IndexOrLedger::Index(meta.index) key_principal symbol=meta.symbol.clone()/> }
+                    view! { <Transactions source=IndexOrLedger::Index(meta.index) key_principal symbol=meta.symbol.clone() decimals/> }
                 } else {
                     view! {
                         <Transactions
                             source=IndexOrLedger::Ledger(meta.ledger)
                             key_principal=None
                             symbol=meta.symbol.clone()
+                            decimals
```

Comment:
Unrelated to this PR, but from what I can see key_principal depends on source?
You should update `IndexOrLedger::Index` variant so it takes key_principal as well.